builtin/providers/librato/resource_librato_metric.go

```diff
@@ -0,0 +1,432 @@
+		}
+		if v, ok := attributeDataMap["created_by_ua"].(string); ok && v != "" {
+			attributes.CreatedByUA = *librato.String(v)
+		}
+		if v, ok := attributeDataMap["display_stacked"].(bool); ok {
+			attributes.DisplayStacked = *librato.Bool(v)
+		}
+		if v, ok := attributeDataMap["gap_detection"].(bool); ok {
+			attributes.GapDetection = *librato.Bool(v)
+		}
+		if v, ok := attributeDataMap["aggregate"].(bool); ok {
+			attributes.Aggregate = *librato.Bool(v)
+		}
+
+		metric.Attributes = attributes
+		fullMetric.Attributes = attributes
+	}
+
+	log.Printf("[INFO] Updating Librato metric: %v", structToString(metric))
+	log.Printf("[INFO] Librato fullMetric: %v", structToString(fullMetric))
+
+	_, err = client.Metrics.Edit(metric)
+	if err != nil {
+		return fmt.Errorf("Error updating Librato metric: %s", err)
+	}
+
+	log.Printf("[INFO] Updated Librato metric %s", metricID)
+
+	// Wait for propagation since Librato updates are eventually consistent
+	wait := resource.StateChangeConf{
+		Pending:                   []string{fmt.Sprintf("%t", false)},
+		Target:                    []string{fmt.Sprintf("%t", true)},
+		Timeout:                   5 * time.Minute,
+		MinTimeout:                2 * time.Second,
+		ContinuousTargetOccurence: 5,
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[INFO] Checking if Librato Metric %s was updated yet", metricID)
+			changedMetric, _, getErr := client.Metrics.Get(metricID)
+			if getErr != nil {
+				return changedMetric, "", getErr
+			}
+			isEqual := reflect.DeepEqual(*fullMetric, *changedMetric)
+			log.Printf("[INFO] Updated Librato Metric %s match: %t", metricID, isEqual)
+			return changedMetric, fmt.Sprintf("%t", isEqual), nil
+		},
+	}
+
+	_, err = wait.WaitForState()
+	if err != nil {
+		log.Printf("[INFO] ERROR - Failed updating Librato Metric %s: %s", metricID, err)
+		return fmt.Errorf("Failed updating Librato Metric %s: %s", metricID, err)
+	}
+
+	return resourceLibratoMetricRead(d, meta)
+}
+
+func resourceLibratoMetricDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*librato.Client)
+
+	id := d.Id()
+
+	log.Printf("[INFO] Deleting Metric: %s", id)
+	_, err := client.Metrics.Delete(id)
+	if err != nil {
+		return fmt.Errorf("Error deleting Metric: %s", err)
+	}
+
+	log.Printf("[INFO] Verifying Metric %s deleted", id)
+	resource.Retry(1*time.Minute, func() *resource.RetryError {
+
+		log.Printf("[INFO] GETing Metric %s", id)
+		_, _, err := client.Metrics.Get(id)
+		if err != nil {
+			if errResp, ok := err.(*librato.ErrorResponse); ok && errResp.Response.StatusCode == 404 {
+				log.Printf("[INFO] GET returned a 404 for Metric %s\n", id)
+				return nil
+			}
+			log.Printf("[INFO] non-retryable error attempting to Get metric: %s", err)
+			return resource.NonRetryableError(err)
+		}
+
+		log.Printf("[INFO] retryable error attempting to Get metric: %s", id)
+		return resource.RetryableError(fmt.Errorf("metric still exists"))
+	})
```

Comment:
Not handling the possible error returned from `resource.Retry`. Can use: 

```
retryErr := resource.Retry()
if retryErr != nil { 
  return fmt.Errorf("Error deleting librato metric: %s", err)
}
```
This will catch if the retry function timed out during the delete.